pkg/cache/bigquerycache/bigquery.go

```diff
@@ -0,0 +1,223 @@
+package bigquerycache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	uuid2 "github.com/google/uuid"
+	"github.com/openshift/sippy/pkg/apis/cache"
+	sippybq "github.com/openshift/sippy/pkg/bigquery"
+	"github.com/openshift/sippy/pkg/cache/compressed"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+)
+
+// https://cloud.google.com/bigquery/quotas#streaming_inserts
+// Maximum row size 	10 MB 	Exceeding this value causes invalid errors.
+// HTTP request size limit 	10 MB Exceeding this value causes invalid errors.
+const (
+	cachedTable     = "cached_data"
+	partitionColumn = "modified_time"
+	chunkSize       = 7000000 // ~7MB to stay under the max row limit
+)
+
+type Cache struct {
+	client     *sippybq.Client
+	dataset    string
+	readOnly   bool
+	expiration time.Duration
+}
+
+func NewBigQueryCache(client *sippybq.Client, dataset string, expiration time.Duration, readOnly bool) (cache.Cache, error) {
+	c := &Cache{
+		client:     client,
+		dataset:    dataset,
+		readOnly:   readOnly,
+		expiration: expiration,
+	}
+
+	return &compressed.Cache{Cache: c}, nil
+}
+
+type CacheRecord struct {
+	Key        string    `bigquery:"key"`
+	UUID       string    `bigquery:"uuid"`
+	Modified   time.Time `bigquery:"modified_time"`
+	Expiration time.Time `bigquery:"expiration"`
+	Data       []byte    `bigquery:"data"`
+	ChunkIndex int       `bigquery:"chunk_index"`
+}
+
+func (c Cache) Get(ctx context.Context, key string) ([]byte, error) {
```

Comment:
It would be very helpful here to instrument this and Set. Prometheus counters for cache hits and missed on warm vs cold. Histograms for the size of the keys being set if you could predict ~10 bucket sizes. I'd settle for logging though. 

Being able to see how it's behaving would be quite beneficial.

Reply:
Added some for hits, misses and sets.  Data sizes are in the logs now but we can pursue histograms if there is value.
Sample values from deleting redis pod and also forcing refresh
```
# HELP sippy_persistent_backend_cache_get Number of persistent cache get that query the backend cache.
# TYPE sippy_persistent_backend_cache_get counter
sippy_persistent_backend_cache_get 1
# HELP sippy_persistent_cache_backend_set Number of persistent cache backend set calls.
# TYPE sippy_persistent_cache_backend_set counter
sippy_persistent_cache_backend_set 5
# HELP sippy_persistent_cache_get Number of persistent cache get requests.
# TYPE sippy_persistent_cache_get counter
sippy_persistent_cache_get 47
# HELP sippy_persistent_warm_cache_miss Number of persistent cache gets that were not in the warm cache.
# TYPE sippy_persistent_warm_cache_miss counter
sippy_persistent_warm_cache_miss 15
```